source/common/router/router.cc

```diff
@@ -478,17 +494,17 @@ void Filter::onUpstreamReset(UpstreamResetType type,
       body = "upstream connect error or disconnect/reset before headers";
     }
 
-    chargeUpstreamCode(code, upstream_host,
-                       reset_reason.valid() &&
-                           reset_reason.value() == Http::StreamResetReason::Overflow);
+    const bool dropped =
+        reset_reason.valid() && reset_reason.value() == Http::StreamResetReason::Overflow;
+    chargeUpstreamCode(code, upstream_host, dropped);
     // If we had non-5xx but still have been reset by backend or timeout before
     // starting response, we treat this as an error. We only get non-5xx when
     // timeout_response_code_ is used for code above, where this member can
     // assume values such as 204 (NoContent).
     if (!Http::CodeUtility::is5xx(enumToInt(code))) {
       upstream_host->stats().rq_error_.inc();
     }
-    Http::Utility::sendLocalReply(*callbacks_, stream_destroyed_, code, body);
+    sendLocalReply(code, body, dropped);
```

Comment:
I think there are some other sites for `sendLocalReply` in this file, but maybe not all apply.